src/cluster_asm.c

```diff
@@ -906,18 +930,22 @@ void asmRdbChannelSyncWithSource(connection *conn) {
 char *asmSendSlotRangesSync(connection *conn, asmTask *task) {
     /* Prepare CLUSTER SYNCSLOTS RANGES command */
     serverAssert(task->slot_ranges->num_ranges <= CLUSTER_SLOTS);
-    int argc = task->slot_ranges->num_ranges*2 + 3;
+    int argc = task->slot_ranges->num_ranges*2 + 5;
     char **args = zcalloc(sizeof(char*) * argc);
     size_t *lens = zcalloc(sizeof(size_t) * argc);
 
     args[0] = "CLUSTER";
     args[1] = "SYNCSLOTS";
-    args[2] = "RANGES";
+    args[2] = "IMPORT";
+    args[3] = task->id;
+    args[4] = "RANGES";
     lens[0] = strlen("CLUSTER");
     lens[1] = strlen("SYNCSLOTS");
-    lens[2] = strlen("RANGES");
+    lens[2] = strlen("IMPORT");
```

Comment:
I added this not to add ID into RANGES block like "RANGES <ID> <start> <slot> .." 
Let me know if you have a better idea.

Reply:
ok, lets use `RANGES <ID> <start> <slot>`

Reply:
The way I see, replication starts with RANGES message. So, I feel like task ID and slot ranges should be part of it.